lib/models/product/models.dart

```diff
@@ -19,6 +19,37 @@ class Product with _$Product {
       _$ProductFromJson(json);
 }
 
+enum Category {
+  none('none'),
+  electronics('electronics'),
+  jewelery('jewelery'),
+  @JsonValue("men's clothing")
+  mensClothing("men's clothing"),
+  @JsonValue("women's clothing")
+  womensClothing("women's clothing");
+
+  final String label;
+
+  const Category(this.label);
+}
+
+extension CategoryExtension on Category {
+  static Category categoryFromString(String category) {
+    switch (category) {
```

Comment:
Replace extension with factory constructor named as 'fromJson(String json)' inside enum.

Reply:
I somehow thought that the fromJson is not generated with json_serializable for enums...